lib/rucio/core/rule.py

```diff
@@ -3220,3 +3280,30 @@ def get_scratch_policy(account, rses, lifetime, session=None):
         if [rse for rse in rses if list_rse_attributes(rse_id=rse['id'], session=session).get('type') == 'SCRATCHDISK']:
             lifetime = 60 * 60 * 24 * scratchdisk_lifetime - 1
     return lifetime
+
+
+@read_session
+def get_rules_for_rse(rse_id, session=None):
+    """
+    Get all rules associated with one rse.
+
+    :param rse_id: The associated rse.
+    :param session: The database session in use.
+    :returns: A list with all the associated rules. The rules only include their ids and rse_expression.
+    """
+    try:
+        query = session.query(models.ReplicationRule.id, models.ReplicationRule.rse_expression) \
+                       .join(models.DataIdentifierAssociation,
+                             and_(
+                                 models.DataIdentifierAssociation.scope == models.ReplicationRule.scope,
+                                 models.DataIdentifierAssociation.name == models.ReplicationRule.name)) \
+                       .join(models.RSEFileAssociation,
+                             and_(
+                                 models.RSEFileAssociation.scope == models.DataIdentifierAssociation.child_scope,
+                                 models.RSEFileAssociation.name == models.DataIdentifierAssociation.child_name)) \
+                       .filter(models.RSEFileAssociation.rse_id == rse_id) \
+                       .distinct()
```

Comment:
I'm not sure if this could lead to performance issues, I would like to have the `filter` as early as possible, but the query must return the `ReplicationRule` entries.